### PR TITLE
T05 child menu UI

### PR DESCRIPTION
**Pull request by Digital-Brendan**
I was tasked with creating a simple UI for the menu that presents categories to the child, and implement testing.

- Scrollable column of clickable categories
- Each row is responsive (new image previews can easily be added)
- Once connected to other views, clicking on a category/row will navigate user to it
- Future plans to add background colour customisation (Yousef's job?)

I struggled to think of what other tests would be useful considering, right now, the only thing that this card asked for was buttons - any additions welcome.

Thanks!

### CODE REVIEW

**Comment by Digital-Brendan** [comment]
You may see that, for my first few commits, there's a discrepancy with the email I used (I changed to a new workstation and forgot to update `git config --global`). Hopefully this doesn't skew things too much on my TeamFeedback page.

**Comment by Digital-Brendan** [comment]
Also you will be aware that I used FlutterFlow to create the simple design. Please know that I won't be using this in the future due to its unnecessarily cumbersome overhead - I apologise for this

**Review by Ali-Alkhars — CHANGES_REQUESTED** [review]
Hey, the UI looks pretty cool! Well done.

**However, I have the following questions/change requests:**
- The boxes now don't take an image. They are just white boxes. You should convert them to a card with an image as a child (so you could do the black borders).
- I noticed that categories and items are all hardcoded, why is that? I think you should convert it into a "model" of some sort, so when implementing the back end you could just use the model and supplement it with the images urls.
- Are the categories clickable? I couldn't tell. If not, then you should make them clickable but leave the onTap implementation empty.
- You should add general comments on top of your pages, explaining what they do
- From what I understood, you only used FlutterFlow for colouring. I don't really think we need it especially since all the colours will be dynamically changable. It also adds unnecessary files. You could delete it.

**Comment by skellyhoe** [comment]
I'll take a look at this when changes are made

**Comment by Digital-Brendan** [comment]
In response to Ali's useful feedback, I refactored pretty much everything after relying on FlutterFlow in the beginning which ended up creating poor code. 

Changes:
- The boxes now take images (though currently use placeholder images until connected to backend - this task was only UI)
- Categories aren't hardcoded. You specify the number of categories (rows) and what image URLs to use, and they are created on screen in a scrollable column
- The categories are now clickable
- Comments have been added where appropriate
- FlutterFlow bloat has been removed

I hope this solves some of the issues that were raised and that the code is now at a satisfactory standard.

Thanks!

@Ali-Alkhars @skellyhoe

**Comment by Digital-Brendan** [comment]
Also, please note that the errors in the integration testing files aren't my fault (I don't think) :)

**Review by AymenBerb — COMMENTED** [review]
Hello Brendan, you have pulled the master branch, but haven't added your work coherently to the behaviour of the app. Can you please keep the same main file as in 'master', and add an access to your board from the 'Child board' button in the side menu of the admin board page. 

Also, I think it would be great idea to have a small horizontal line separating the categories. (To be asked in the gc)

**Review by skellyhoe — APPROVED** [review]
Fixed your .gitignore and removed the files you missed. Looks good

File: lib/pages/child_menu.dart [hunk 1]
```diff
@@ -0,0 +1,215 @@
+import 'package:flutter/material.dart';
+import 'child_board/child_board_interface.dart';
+
+// The child menu is formed essentially by creating a column of rows,
+// with rowConfigs outlining the category's title and images
+// (1st image is category image and those after are the smaller previews
+//
+
+class CustomizableColumn extends StatelessWidget {
+  // const CustomizableColumn({Key? key, required this.rowConfigs}) : super(key: key);
```
Comment (Ali-Alkhars) [inline lib/pages/child_menu.dart]: Remove commented out code

File: lib/pages/child_menu.dart [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        itemCount: rowConfigs.length,
+        separatorBuilder: (context, index) {
+          return Divider(height: 2);
+        },
+      ),
+    );
+  }
+}
+
+class CustomizableRow extends StatelessWidget {
```
Comment (Ali-Alkhars) [inline lib/pages/child_menu.dart]: You can move this class into a separate file in the widgets folder (and all the classes below)

**Review by Ali-Alkhars — CHANGES_REQUESTED** [review]

**Comment by skellyhoe** [comment]
Summary of changes requested by me in the gc:

- Separate customizableRow into smaller widgets.
- CustomizableColumn should return a scaffold instead of a listview.
- Remove generated plugin files.

**Comment by Digital-Brendan** [comment]
Please find the final version of this UI, thanks.

**Review by Ali-Alkhars — APPROVED** [review]
Well done Brendan! Thanks for doing all the requested changes.

**Pull request merged**